WIP: pre-pull-image-curl (#1691)
codenvy/che-docs: modules/administration-guide/examples/che-prepulled-images-table.adoc

|che-nodejs10-ubi |quay.io/eclipse/che-nodejs10-ubi:{prod-ver} |{prod-ver}

|che-quarkus |quay.io/eclipse/che-quarkus:{prod-ver} |{prod-ver}
|===

[NOTE]
====
The table of recommended images to pre-pull is updated once per release and might vary from the most updated build at the time.

To get the latest build:

. Use the following scriptlet:
+
[subs="+attributes,+quotes"]
----
$➔ curl -sSLo- https://raw.githubusercontent.com/eclipse/che-operator/master/olm/eclipse-che-preview-openshift/deploy/olm-catalog/eclipse-che-preview-openshift/{che-ver}/eclipse-che-preview-openshift.v{che-ver}.clusterserviceversion.yaml | yq -r '.spec.relatedImages[].image' | sort | uniq
----
+

> . Exclude redundant, unused digests, and unsupported images for OpenShift pre-pulling such as:

Reviewer: This instruction is not going to be something a user can obviously/easily do.

To make it easier, perhaps we should instead use:

```
$➔ curl -sSLo- https://raw.githubusercontent.com/eclipse/che-operator/master/olm/eclipse-che-preview-openshift/deploy/olm-catalog/eclipse-che-preview-openshift/{che-ver}/eclipse-che-preview-openshift.v{che-ver}.clusterserviceversion.yaml | \
  yq -r '.spec.relatedImages[]
```

And then provide a snippet like:
```
...
{
  "name": "che-theia-7.19.2",
  "image": "quay.io/eclipse/che-theia@sha256:11b5fe33edd72df22f76d7933ba79029051f4c0c6adc6c1a91f2dfe4228455c9"
}
{
  "name": "che-theia-7.20.0",
  "image": "quay.io/eclipse/che-theia@sha256:92a8bac94e446f1f933d1a92bc96dace9450c9f4efdef5539a429a37d14be904"
}
{
  "name": "che-theia-7.20.1",
  "image": "quay.io/eclipse/che-theia@sha256:a56428ee28ceefca2a1cba51093f6015e0eee82659fec786f2e1fdc939fdb856"
}
...
```

With the instruction:

```
. Where multiple versions of the same image are listed, only the latest one needs to be pre-pulled. 

Note also that images like machine-exec, images without the `sleep` command and images that mount volumes cannot be pre-pulled, and thus need to be pulled by hand with `skopeo copy --all ...`.
```

+
* machine-exec image
* images without a sleep command installed
* images that mount volumes in the docker file
+

> For additional information, see link:https://github.com/eclipse/che/issues/17562[GitHub Issue 17562]

Reviewer: Note that old versions of the same image will be removed soon. See https://github.com/eclipse/che-plugin-registry/pull/672 and https://github.com/eclipse/che/issues/18183

If you wait to merge this PR until after Che 7.22.0 is live, the instructions will be simpler.

Author: OK, will prepare your suggestion and wait for the 7.22.0

|che-nodejs10-ubi |quay.io/eclipse/che-nodejs10-ubi:{prod-ver} |{prod-ver}

|che-quarkus |quay.io/eclipse/che-quarkus:{prod-ver} |{prod-ver}
|===

[NOTE]
====

> The table of recommended images to pre-pull is updated once per release and might vary from the most updated build at the time.

Reviewer 2: Let's get rid of the table completely and only offer the procedure below.

To get the latest build:

> . Install the dependencies:

Reviewer 2: These should be prerequisites.

+
----
# dnf install -y jq python-pip
----
+
----
# pip install yq
----

> . Use the following scriptlet:

Reviewer 2: ```suggestion
.Procedure

. Use the following scriptlet:
```

+
[subs="+attributes,+quotes"]
----
$ curl -sSLo- https://raw.githubusercontent.com/eclipse/che-operator/master/olm/eclipse-che-preview-openshift/deploy/olm-catalog/eclipse-che-preview-openshift/{che-ver}/eclipse-che-preview-openshift.v{che-ver}.clusterserviceversion.yaml | \
  yq -r '.spec.relatedImages[]
----
+

> ----

Reviewer 2: ```suggestion
[source,json]
----
```

> ...

Reviewer 2: ```suggestion
#[...]
```

{
  "name": "che-theia-7.19.2",
  "image": "quay.io/eclipse/che-theia@sha256:11b5fe33edd72df22f76d7933ba79029051f4c0c6adc6c1a91f2dfe4228455c9"
}
{
  "name": "che-theia-7.20.0",
  "image": "quay.io/eclipse/che-theia@sha256:92a8bac94e446f1f933d1a92bc96dace9450c9f4efdef5539a429a37d14be904"
}
{
  "name": "che-theia-7.20.1",
  "image": "quay.io/eclipse/che-theia@sha256:a56428ee28ceefca2a1cba51093f6015e0eee82659fec786f2e1fdc939fdb856"
}

> ...

Reviewer 2: ```suggestion
#[...]
```

----
+

> Following images from the list cannot be pre-pulled, therefore need to be pulled manually with `skopeo copy --all __<...>__`:

Reviewer 2: ```suggestion
The following images from the list cannot be pre-pulled and need to be pulled manually using:

[subs="+quotes"]
----
$ skopeo copy --all _<image>_
----
```

+

> * machine-exec image

Reviewer 2: List all these images by name (without versions) to make it clear, which ones are they.
